feline/src/main/java/com/spotify/feline/Feline.java

```diff
@@ -32,7 +31,7 @@
 /** Detects blocking calls to @link CompletableFuture and notifies registered consumers. */
 public class Feline {
 
-  private static final Map<String, Set<String>> allowances = new ConcurrentHashMap<>();
+  private static final AllowancesTransformer allowancesTransformer = new AllowancesTransformer();
```

Comment:
I think this falls under the same problem sadly, because it's a static field within `Feline`. I try something later, but I think we will need to remove this static and move to inside the `AllowancesTransformer`

Reply:
created #20 to show what I meant

Reply:
This field is explicitly passed into Agent builder, so ByteBuddy will never have to circle back to this field (like it was in case of lambda), so I actually think this would work 🤔

Reply:
isn't the problem to access any static field? Although I see what you mean, as the lambda won't try to access the `allowances` it still trying to access this `allowancesTransformer`, which will be locked until the static initiation stops. Moving completely to another class removes the "competition" for static resources from this class

Reply:
I think if we expand lambda to actual class instance it would be more visible, it will be something like this:
```
.type(new ElementMatcher<TypeDescription>() {
    public boolean matches(final TypeDescription typeDescription) {
        return Feline.allowances.containsKey(it.getName());
    }
})
```

Which basically means that when ByteBuddy will be evaluating the lambda - the lambda will need to access the `Feline` class again.

In the Matt's code, there is no reference back to `Feline` when the `ElementMatcher` is evaluated, so that I believe should work 🤔

Reply:
what @TheIndifferent talks about here is exactly my line of thinking. The current code passes a reference to a static method/field to `AgentBuilder.type(..)` (which happens on the `main` thread) and when another thread does something that causes the bytebuddy transformer to be invoked it has to call/access the field/method in `Feline`. 

The `javap` output for the class might help make that more clear, I posted that in https://github.com/spotify/futuristic-feline/issues/18#issuecomment-1205204282

Reply:
alright, agreed. As we can't really replicate I will try to release a release candidate internally and see if any of the people that had the problem is willing to test it

Reply:
I don't have the powers to merge this, would you mind doing it @mattnworb?